lib/bolt_server/file_cache.rb

```diff
@@ -166,8 +166,11 @@ def expire(purge_ttl)
       expired_time = Time.now - purge_ttl
       @cache_dir_mutex.with_write_lock do
         Dir.glob(File.join(@cache_dir, '*')).select { |f| File.directory?(f) }.each do |dir|
-          if (mtime = File.mtime(dir)) < expired_time
-            @logger.debug("Removing #{dir}, last used at #{mtime}")
+          next unless (mtime = File.mtime(dir)) < expired_time
+          @logger.debug("Removing #{dir}, last used at #{mtime}")
+          if dir == tmppath
+            FileUtils.rm_rf(Dir.glob("#{tmppath}/*"))
```

Comment:
It was not immediately clear to me if it is possible that anything would be left in the tmp dir to delete? I see we use a `Tempfile` i'm not sure exactly how those are handled.

Reply:
I think it should be fine wihtout this. I think this causes a race since the tmpdir operations don't use the mutex so an in progress download would get deleted.

Reply:
Chose to skip deleting anything in `tmppath` given that tmpdir operations don't use the mutex and should always be cleaned.